### DISCUSSION TITLE
Pluggable SerDe - topic and proto class mapping UI

### DISCUSSION THREAD

**Discussion by adformkyadav** [provectus/kafka-ui#3332] [Ideas]
Hi,

The pluggable SerDe is a great feature of kafka-ui which I use.

Here is how I use it:
I have lots of topics and contracts for their messages which can keep on updating quite frequently. So what I did was compiled the contract protobuf files and package it into jar and store in a repository. So whenever a contract need to be updated I just publish a new release of contract jar.

I plugged a class for serde in kafka-ui which do followings:
1. Download the latest version of contract jar from repository
2. Load the jar classes in jvm 
3. Provide deserialiser for the given topic key/value whenever request for one comes
4. I use a static config file to store the mapping of topic_key/value->deserialiser class

So far it works fine, but problem is when a new topic is added and I have to update this static config file and restart the kafka-ui.

To avoid this restarts, what I could think of is, store the config mappings the a DB table and load it in memory whenever request for deserialiser for a topic comes. That way I will always have latest mapping for any topic.
But for this I have to provide a mechanism to update the mapping from somewhere (I don't want to update it directly from DB for obvious reasons)
I can create an UI from where I can do it, but I was thinking if kafka-ui could provide such mechanism to extent the its UI for such cases? or is there any way to handle this particular use case?

**Comment by Haarolean** [comment]
Might be solvable within #3501
